georocket-common/src/main/java/io/georocket/query/DefaultQueryCompiler.java

```diff
@@ -262,16 +265,31 @@ public void enterString(StringContext ctx) {
         }
       }
     }
-    
+
     @Override
-    public void enterKeyvalue(KeyvalueContext ctx) {
+    public void enterLt(LtContext ctx) {
       currentKeyvalue = new JsonObject();
+      currentKeyvalue.put("comp", Comparator.LT.name());
     }
-    
+
+    @Override
+    public void enterGt(GtContext ctx) {
+      currentKeyvalue = new JsonObject();
+      currentKeyvalue.put("comp", Comparator.GT.name());
+    }
+
+    @Override
+    public void enterEq(EqContext ctx) {
+      currentKeyvalue = new JsonObject();
+      currentKeyvalue.put("comp", Comparator.EQ.name());
+    }
+
     @Override
     public void exitKeyvalue(KeyvalueContext ctx) {
```

Comment:
Typically not, unless you run into the same problem I had with `string` and `expr` because a string is also an expression but I didn't want to visit it all the time, so I had to split `enterExpr` up into the individual rules implementing `expr`.